addon/authenticators/oauth2-password-grant.js

```diff
@@ -164,11 +161,19 @@ export default BaseAuthenticator.extend({
       }
       this.makeRequest(serverTokenEndpoint, data).then((response) => {
         run(() => {
+          this._validate(reject, response, 'access_token');
+          const refreshAccessTokens = this.get('refreshAccessTokens');
+          if (refreshAccessTokens) {
+            this._validate(reject, response, 'expires_in');
+            this._validate(reject, response, 'refresh_token');
```

Comment:
If `{ error: 'invalid_grant' }` nothing else should be in the response actually. But in this case the response code would also be 422 or so so that the request's success handler should never be called anyway. I think we only need to validate successful server responses as erroneous responses will already reject the returned promise anyway.